2024/incoming/being-lazy-with-github-actions.pod

```diff
@@ -0,0 +1,198 @@
+Naughty list herself.
+
+"I need help," she muttered to herself during one particularly long debugging
+session. She decided to take a break, grabbed a coffee from the machine and
+started idly browsing r/elftech. Immediately, an article about GitHub Actions
+caught her attention.
+
+=head2 What Are GitHub Actions?
+
+Pixie had vaguely heard of GitHub Actions before, but she’d always assumed
+it was for the tech-savvy elves in charge of JavaScript or Rust projects.
+The article made it seem thatGitHub Actions were just as useful for Perl
+developers, and that it could automate much of her workflow.
+
+Intrigued, Pixie decided to dive in.
+
+=head2 A First Workflow: Automating Tests
+
+Pixie started with the basics: automating the test suite for NaughtyOrNice.
+The app already had a robust set of tests written with Test::More, but running
+them manually was tedious.
+
+Pixie created her first GitHub Actions workflow file, .github/workflows/test.yml:
+
+=begin code
+
+    name: Perl Test Suite
+
+    on:
+      push:
+        branches:
+          - main
+      pull_request:
+
+    jobs:
+      test:
+        runs-on: ubuntu-latest
+
+        steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+
+        - name: Set up Perl
+          uses: shogo82148/actions-setup-perl@v1
+          with:
+            perl-version: '5.40'
+
+        - name: Install dependencies
+          run: |
+            cpanm --quiet --installdeps .
+
+        - name: Run tests
+          run: prove -lr t/
+
+=end code
+
+Pixie was impressed. With just a few lines of YAML, she could trigger her
+tests automatically whenever someone pushed code to the main branch or opened
+a pull request. No more manual testing! She could focus on writing code
+instead of running it.
+
+=head2 Adding Deployment: Automating the NaughtyOrNice Rollout
+
+With her test suite humming along, Pixie turned her attention to deployments.
+Previously, deploying the app involved copying files to the production
+server with a combination of scp commands and festive hope.
+
+She created another workflow file, .github/workflows/deploy.yml:
+
+=begin code
+
+    name: Deploy to Production
+
+    on:
+      workflow_dispatch:
+
+    jobs:
+      deploy:
+        runs-on: ubuntu-latest
+
+        steps:
+        - name: Checkout code
+          uses: actions/checkout@v3
+
+        - name: Deploy to server
+          env:
+            RSYNC_PASSWORD: ${{ secrets.RSYNC_PASSWORD }}
+          run: |
+            rsync -avz --delete ./ naughtyornice@production-server:/var/www/naughtyornice/
+
+=end code
+
+By adding a secret to the GitHub repository for her server credentials, Pixie
+automated the entire deployment process. The `workflow_dispatch` key in the
+`on` section of the workflow added a "Run" button to the "Actions" tab in the
+GitHub repo, and just pressing that would deploy the application to the
+production server. No more late nights. No more missed sleigh rides. Pixie
```

Comment:
There are both backticks and double-quotes used here, neither of which are significant in POD. Should one or both be replaced by POD syntax?

Reply:
Oops. Yes. The problem of switching between Pod and Markdown. Thanks for spotting it. Fixed.